utils/parallelisation/group.go

```diff
@@ -212,12 +215,18 @@ type IExecutor interface {
 type IExecutionGroup[T any] interface {
 	IExecutor
 	RegisterFunction(function ...T)
+	// RegisterFunctions registers a sequence of functions to the group.
+	RegisterFunctions(function iter.Seq[T])
+	// CopyFunctions copies the registered functions to group g
+	CopyFunctions(g IExecutionGroup[T])
 	Len() int
+	// Clone returns a new execution group with all the same functions being registered.
```

Comment:
Can we provide some context on what happens if a group has already been executed.

Particularly in cases where you have used `OnlyOnce` as an option.

Should `Execute` be idempotent across clones? Or should it be like a context?